pydocstringformatter/formatting/formatter.py

```diff
@@ -16,6 +16,20 @@ def _treat_string(self, tokeninfo: tokenize.TokenInfo, _: int) -> str:
         return new_string
 
 
+class CapitalizeFirstLetter(StringFormatter):
+    """Capitalize the first letter of the docstring if appropriate."""
+
+    name = "capitalize-first-letter"
+    SNAKE_CASE = re.compile(r"([^\W\dA-Z]+_[^\WA-Z]*)$")
+
+    def _treat_string(self, tokeninfo: tokenize.TokenInfo) -> str:
+        new = tokeninfo.string[3:].lstrip()
```

Comment:
Thinking about this: should the `lstrip` be in this formatter? Perhaps not right? This should simply return `tokeninfo.string[:3] + tokeninfo.string[4].upper() + tokeninfo.string[5:]` while we could also add a `DocstringWhitespaceStripperFormatter` (what a name...).

Reply:
Added benefit of this would be that this checker wouldn't be order dependent anymore and we can keep the order dependent formatters to a minimum and self-explanatory. ie: it's logical that a whitespace stripper should go first, it is not immediately logical why a capitaliser should go first.

Reply:
We would need to detect the first letter in case there's multiple spaces, it would impact performance a little. But I agree that doing one thing only is important.

Reply:
Hm, what if a user disabled `trim-whitespace-formatter` but does enable `capitalise-first-letter-formatter`. Should we then "capitalise the whitespace" or capitalise the first letter we find?

Reply:
Capitalizing the whitespace does nothing so I think it's not intuitive and we should capitalize the first letter ?

Reply:
Yeah I guess. Can we do that with `r"\['"]+\s*(\w)"`? That should get the first letter after any possible whitespaces and removes the need to strip etc.